src/librustc_parse/parser/path.rs

```diff
@@ -471,38 +471,43 @@ impl<'a> Parser<'a> {
         Ok(self.mk_ty(span, ast::TyKind::Err))
     }
 
+    /// Parse a const argument, i.e. <3>, omitting the angle brackets.
+    pub(super) fn parse_const_arg(&mut self) -> PResult<'a, GenericArg> {
```

Comment:
Please return `PResult<'a, AnonConst>` or instead and wrap it `using Ok(GenericArg::Const(AnonConst { id: ast::DUMMY_NODE_ID, value: self.parse_const_arg()? }))` in `parse_generic_arg`.

We really shouldn't use `GenericArg` for the default value of const params.